examples/examples_yaml_test.go

```diff
@@ -187,6 +187,24 @@ func TestElasticacheReplicationGroupUpgrade(t *testing.T) {
 	testProviderUpgrade(t, filepath.Join("test-programs", "elasticache-replication-group"), nil)
 }
 
+func TestS3BucketToBucketUpgrade(t *testing.T) {
```

Comment:
Added these two tests which test both types of upgrades.

- `Bucket (legacy) => Bucket`
- `BucketV2 => Bucket`